android/src/org/mavlink/qgroundcontrol/QGCProber.java

```diff
@@ -0,0 +1,54 @@
+package org.mavlink.qgroundcontrol;
+
+import com.hoho.android.usbserial.driver.ProbeTable;
+import com.hoho.android.usbserial.driver.UsbSerialProber;
+import com.hoho.android.usbserial.driver.CdcAcmSerialDriver;
+import org.mavlink.qgroundcontrol.QGCUsbId;
+
+// implements UsbSerialProber?
+class QGCProber
+{
+    static UsbSerialProber getQGCProber()
+    {
+        final ProbeTable customTable = new ProbeTable();
+
+        customTable.addProduct(QGCUsbId.VENDOR_PX4, QGCUsbId.DEVICE_PX4FMU_V1, CdcAcmSerialDriver.class);
+        customTable.addProduct(QGCUsbId.VENDOR_PX4, QGCUsbId.DEVICE_PX4FMU_V2, CdcAcmSerialDriver.class);
+        customTable.addProduct(QGCUsbId.VENDOR_PX4, QGCUsbId.DEVICE_PX4FMU_V3, CdcAcmSerialDriver.class);
+        customTable.addProduct(QGCUsbId.VENDOR_PX4, QGCUsbId.DEVICE_PX4FMU_V4, CdcAcmSerialDriver.class);
+        customTable.addProduct(QGCUsbId.VENDOR_PX4, QGCUsbId.DEVICE_PX4FMU_V4PRO, CdcAcmSerialDriver.class);
```

Comment:
Do we need to add the whole list of FMU versions here?

Reply:
So this list (the entire custom table) is the only part I really have no idea about, as I can't test every device. The cube I used worked without the use of the customTable entirely. I honestly am not sure what to put in it or not.